appengine/standard_python37/redis/main_test.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+import main
+
+
+def test_index():
+    try:
+        main.redis_client.set('counter', 0)
+    except Exception:
+        pytest.skip('Redis is unavailable.')
+
+    main.app.testing = True
+    client = main.app.test_client()
+
+    r = client.get('/')
+    assert r.status_code == 200
+    assert '1' in r.data.decode('utf-8')
```

Comment:
Nit: what is the format of `r.data`? Is there any chance of spurious `1`s or `2`s occurring in it?

Reply:
No, this test is copied from Flex and we didn't have any issues. The format is an integer but it'll be starting from zero every time, and the feature that determines that is not a component under test.